google/import_google_service_account_test.go

```diff
@@ -0,0 +1,44 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"os"
+)
+
+func TestAccGoogleServiceAccount_importBasic(t *testing.T) {
+	t.Parallel()
+
+	resourceName := "google_service_account.acceptance"
+	project := os.Getenv("GOOGLE_PROJECT")
```

Comment:
Is there a reason why you decided to set the project id specifically in this test rather than letting the test read it from the environment? If so mind adding a comment to explain?

Reply:
Good point, project is indeed an optional parameter so it should work with or without it.

Removing it from the import test raises an issue so I guess we should keep both test cases:

```
TF_ACC=1 go test ./google/ -v -run=TestAccGoogleServiceAccount_import.* -timeout 120m                                                                                                     
=== RUN   TestAccGoogleServiceAccount_importBasic                                                                                                                                         
=== RUN   TestAccGoogleServiceAccount_importWithProject                                                                                                                                   
--- FAIL: TestAccGoogleServiceAccount_importBasic (2.66s)                                                                                                                                 
        testing.go:435: Step 0 error: After applying this step, the plan was not empty:                                                                                                   
                                                                                                                                                                                          
                DIFF:                                                                                                                                                                     
                                                                                                                                                                                          
                DESTROY/CREATE: google_service_account.acceptance                                                                                                                         
                  account_id:   "terraform-1nd79xoges" => "terraform-1nd79xoges"                                                                                                          
                  display_name: "terraform-1nd79xoges" => "terraform-1nd79xoges"                                                                                                          
                  email:        "terraform-1nd79xoges@myproject.iam.gserviceaccount.com" => "<computed>"                                                                
                  name:         "projects/myproject/serviceAccounts/terraform-1nd79xoges@myproject.iam.gserviceaccount.com" => "<computed>"
                  project:      "myproject" => "" (forces new resource)
                  unique_id:    "113399152587722893408" => "<computed>"

                STATE:

                google_service_account.acceptance:
                  ID = projects/myproject/serviceAccounts/terraform-1nd79xoges@myproject.iam.gserviceaccount.com
                  account_id = terraform-1nd79xoges
                  display_name = terraform-1nd79xoges
                  email = terraform-1nd79xoges@myproject.iam.gserviceaccount.com
                  name = projects/myproject/serviceAccounts/terraform-1nd79xoges@myproject.iam.gserviceaccount.com
                  project = myproject
                  unique_id = 113399152587722893408
--- PASS: TestAccGoogleServiceAccount_importWithProject (3.91s)
FAIL
exit status 1
FAIL    github.com/terraform-providers/terraform-provider-google/google 3.920s
```

No sure why yet.
Once fixed, I'll add a comment.

Reply:
Adding `Computed: True` to the `google_service_account`'s `project` field ensures the project id is always stored in the state and resolves the issue. Both test cases now pass.

I've also added an extra step and some assertions to the `TestAccGoogleServiceAccount_basic` test case to assert this.